Agregando la traducción al inglés del archivo comandos útiles (#8003)
omegaup/omegaup: frontend/www/docs/Useful-Commands-for-Development.md

Below is a list of commands/scripts that are useful for people involved in development:  

> 

Reviewer: ```suggestion
---
```

Author: Listo

### Run all omegaUp linters
**Command:**
```bash
./stuff/lint.sh
```

> **Description:** Script required to run all code validations available in omegaUp. If not executed directly, it will run automatically when performing `git push`.  

Reviewer: ```suggestion
**Description:**Runs all code validations available in omegaUp. If not executed directly, it will run automatically when performing `git push`.  
```

Author: Listo

**Execution Location:** Outside the Docker container, in the project's root directory. e.g. `ubuntu@pc:~/dev/omegaup$`  

### Generate `.lang` files
**Command:**
```bash
./stuff/lint.sh --linters=i18n fix --all
```
**Description:** Generates `*.lang*` files based on `es.lang`, `en.lang`, and `pt.lang`.  
**Execution Location:** Outside the Docker container, in the project's root directory. e.g. `ubuntu@pc:~/dev/omegaup$`  

### Run all omegaUp tests and validations in PHP
**Command:**
```bash
./stuff/runtests.sh
```

> **Description:** It is currently responsible for running tests in `PHPUnit`, as well as executing the type validator in `MySQL` and `PSALM`.  

Reviewer: ```suggestion
**Description:** Runs tests using `PHPUnit`, as well as executing the type validator in `MySQL` and `PSALM`.  
```

Author: Listo

**Execution Location:** Inside the Docker container, in the root directory.  

### Run Cypress Tests
```bash
npx cypress open
```

> **Description:** It opens the `Cypress Test Runner`, a graphical interface that allows you to interactively run and debug Cypress tests in a browser. It enables you to select and run individual tests, view detailed test results, and inspect any failures in real-time, providing an efficient environment for writing, running, and debugging tests. It could require extra configurations to work properly in the local environment. 

Reviewer: ```suggestion
**Description:** Opens the `Cypress Test Runner`, a graphical interface that allows interactive test execution and debugging. Additional configurations may be required for local environments; `the most common are/is`: 
```

Reviewer: Si se puede, hay que agregar un ejemplo de configuración adicional, o decir donde buscar ayuda (link a documentacion de cypress, etc.)

Author: Listo

**Execution Location:** Outside the Docker container.  

### Reset database to initial state
```bash
./stuff/bootstrap-environment.py --purge
```

> **Description:** A script required to restore the database to its initial state. It also runs a series of API requests to populate the local development environment. With this script, contests, courses, problems, and everything necessary for manual testing can be created. The generated data is stored in the `stuff/bootstrap.json` file.

Reviewer: ```suggestion
**Description:** Restores the database to its initial state and populates the local development environment with data (contests, courses, problems, etc.) for manual testing via a series of API requests. The generated data is stored in the stuff/bootstrap.json file.
```

Author: Listo

**Execution Location:** Inside the Docker container, in the root directory.  

### Run PHP type validators
```bash
find frontend/ \
    -name *.php \
    -and -not -wholename 'frontend/server/libs/third_party/*' \
    -and -not -wholename 'frontend/tests/badges/*' \
    -and -not -wholename 'frontend/tests/controllers/*' \
    -and -not -wholename 'frontend/tests/runfiles/*' \
    -and -not -wholename 'frontend/www/preguntas/*' \
  | xargs ./vendor/bin/psalm \
    --long-progress \
    --show-info=false
```

> **Description:** Command to run type validators on PHP files.  

Reviewer: ```suggestion
**Description:** Runs type validation on PHP files.  
```

Author: Listo

**Execution Location:** Inside the Docker container, in the root directory.  

### Run PHP unit tests for a specific file
```bash
./stuff/run-php-tests.sh frontend/tests/controllers/$MY_FILE.php
```

> **Description:** Command to run unit tests for an individual PHP file. To run all tests, omit the file name.  

Reviewer: ```suggestion
**Description:** Runs unit tests for an individual PHP file. To run all tests, omit the file name.  
```

Author: Listo

**Execution Location:** Inside the Docker container, in the root directory.  

### Apply changes to schema.sql
```bash
./stuff/update-dao.sh
```

> **Description:** Script required to apply changes to the `schema.sql` file when adding a new migration file in `.sql` (It will work until the migration file is committed).  

Reviewer: ```suggestion
**Description:** Applies changes to the `schema.sql` file when adding a new migration file in `.sql`. Works until the migration file is committed. 
```

Author: Listo

**Execution Location:** Inside the Docker container, in the root directory.  

### Apply database migrations locally
```bash
./stuff/db-migrate.py migrate --databases=omegaup,omegaup-test
```

> **Description:** Script required to apply schema changes locally when a new migration file in `.sql` is added.  

Reviewer: ```suggestion
**Description:** Applies local schema changes when a new migration file in `database/*.sql` is added.  
```

Author: Listo

**Execution Location:** Inside the Docker container, in the root directory.  

### Run Vue unit tests
```bash
yarn run test:watch
```

> **Description:** It runs Vue unit tests in "watch" mode. This means that whenever changes are made to test files or the source code files being tested, the tests are automatically rerun to reflect those changes. This makes development easier, as it allows developers to continuously see test results without having to manually rerun them each time a change is made to the code. The command runs within the local development environment of the application, providing an efficient and dynamic workflow during test development.

Reviewer: ```suggestion
**Description:** Runs Vue unit tests in watch mode, automatically rerunning tests when code changes. The command runs within the local development environment of the application, providing an efficient and dynamic workflow during test development.
```

Author: Listo

**Execution Location:** Inside the Docker container, in the root directory.  

### Run a specific Vue unit test file
```bash
./node_modules/.bin/jest frontend/www/js/omegaup/components/$MY_FILE.test.ts
```

> **Description:** Useful command to run a single Vue unit test file.  

Reviewer: ```suggestion
**Description:** Runs a single Vue unit test file.  
```

> **Execution Location:** It works fine inside or outside the docker container.  

Reviewer: ```suggestion
**Execution Location:** Inside or outside the docker container.  
```

Author: Listo

### Restart Docker service
```bash
systemctl restart docker.service
```

> **Description:** Command used to restart the Docker process. It is useful when encountering the following error while running Docker:  

Reviewer: ```suggestion
**Description:** Restarts the Docker service. Useful for resolving errors such as:  
```
